### PR TITLE
Add burn action to NFTs

### PR DESCRIPTION
**Pull request by nachoiacovino**

### CODE REVIEW

File: @3rdweb-sdk/react/hooks/useCommon.ts [hunk 1]
```diff
@@ -375,3 +380,56 @@ export function useAirdropMutation<TContract extends ValidContractInstance>(
...
+                  return cacheKeys.list(contract?.getAddress());
+                }
+                return undefined as never;
+              })
+              .filter((fn) => !!fn),
+          );
+        }
+
+        return invalidate([
+          CacheKeyMap[contractType].list(contract?.getAddress()),
```
Comment (jnsdls) [inline @3rdweb-sdk/react/hooks/useCommon.ts:430]: this needs to also invalidate the `supply` query ("supply" goes down when burning)

File: @3rdweb-sdk/react/hooks/useCommon.ts [hunk 2]
```diff
@@ -375,3 +380,56 @@ export function useAirdropMutation<TContract extends ValidContractInstance>(
...
+      onSuccess: (_data, _variables, _options, invalidate) => {
+        // this should not be possible, but we need to catch it in case it does
+        // if we don't know we just invalidate everything.
+        if (!contractType) {
+          return invalidate(
+            Object.keys(CacheKeyMap)
+              .map((key) => {
+                const cacheKeys = CacheKeyMap[key as keyof typeof CacheKeyMap];
+                if ("list" in cacheKeys) {
+                  return cacheKeys.list(contract?.getAddress());
```
Comment (jnsdls) [inline @3rdweb-sdk/react/hooks/useCommon.ts:421]: same here, needs to invalidate supply query on all contracts in the fail case probably

**Review by jnsdls — CHANGES_REQUESTED** [review]
almost there, just fix invalidation

File: @3rdweb-sdk/react/hooks/useCommon.ts [hunk 3]
```diff
@@ -375,3 +380,59 @@ export function useAirdropMutation<TContract extends ValidContractInstance>(
...
+                  return cacheKeys.supply(contract?.getAddress());
+                }
+                return undefined as never;
+              })
+              .filter((fn) => !!fn),
+          );
+        }
+
+        return invalidate([
+          CacheKeyMap[contractType].list(contract?.getAddress()),
```
Comment (jnsdls) [inline @3rdweb-sdk/react/hooks/useCommon.ts:433]: what about supply here?

**Review by jnsdls — CHANGES_REQUESTED** [review]
need to check if supply exists and invalidate it

File: @3rdweb-sdk/react/hooks/useCommon.ts (hunk 3 above)
Comment (nachoiacovino) [reply @3rdweb-sdk/react/hooks/useCommon.ts:433]: supply only exists in NFTDrop contracts, if I add it here it gives an error on all the other contracts, trying to make sense on how to refresh on Edition in which it shows

Comment (jnsdls) [reply @3rdweb-sdk/react/hooks/useCommon.ts:433]: we definitely *have* supply numbers for them, so find the useQuery for that they're using and make sure they are being invalidated

**Review by jnsdls — APPROVED** [review]

**Pull request merged**